docs/pages/versions/unversioned/sdk/device.md

```diff
@@ -0,0 +1,244 @@
+---
+title: Device
+---
+
+Provide an API to play with the device's system physical information that uniquely associated with the device's hardware and software properties.
+
+## Installation
+
+This API is pre-installed in [managed](../../introduction/managed-vs-bare/#managed-workflow) apps. To use it in a [bare](../../introduction/managed-vs-bare/#bare-workflow) React Native app, follow its [installation instructions](https://github.com/expo/expo/tree/master/packages/expo-device).
+
+## API
+
+```js
+import * as Device from 'expo-device';
+```
+
+### Constants
+
+- `Device.brand: string`
+
+  Gets the device brand. The consumer-visible brand with which the product/hardware will be associated, if any.
+
+  - iOS: `"Apple"`
+  - Android: e.g., `"Xiaomi"`
+
+- `Device.manufacturer: string`
+
+  Gets the actual device manufacturer of the product/hardware.
+
+  - iOS: `"Apple"`
+  - Android: e.g., `"HTC"`
+
+  Click [Here](https://developer.android.com/reference/android/os/Build) to view Android official documentation about the difference between `brand` and `manufacturer`.
+
+- `Device.modelName: string`
+
+  **iOS warning**: The list of the model name is maintained by the community and could miss latest devices, so it is better to use `modelId` since it unique identifies new iOS devices and are always up-to-date. We accept new pull request that helps to update the device modelName list.
+
+  Gets the device model.
+
+  - iOS: e.g., `"iPhone XS Max"`
+  - Android: e.g., `"Pixel 2"`
+
+- `Device.osName: string`
+
+  Gets the device OS name.
+
+  - iOS: e.g., `"iOS"`
+  - Android: e.g., `"Android"`
+
+- `Device.modelId: string` (iOS only)
+
+  Gets the device's mobile device codes types (machine id).
+
+  - iOS: e.g., `"iPhone7,2"`
+
+- `Device.totalMemory: number`
+
+  Gets the device's total memory, in bytes. It means the total memory accessible by the kernel. This is basically the RAM size of the device, not including below-kernel fixed allocations like DMA buffers, RAM for the baseband CPU, etc.
```

Comment:
It returns `@(0)` if we can't grab the total memory on the objective C side. What about android @vivianzzhu91

Reply:
It is a system constant, referenced from here https://developer.android.com/reference/android/app/ActivityManager.MemoryInfo. So I just return this constant and I'm not sure if this would return null or not.